core-bundle/src/Controller/AbstractController.php

```diff
@@ -124,4 +124,19 @@ protected function setCacheHeaders(Response $response, PageModel $pageModel): Re
 
         return $response;
     }
+
+    protected function getPageModel(): PageModel|null
```

Comment:
The changes in this file I would extract into its own PR because this cleanup always makes sense, no matter this PR :)